app/oauth2/oauth2-route.js

```diff
@@ -1,17 +1,27 @@
 const accessTokenRequest = require('./access-token-request');
+const hasAnAttribute = require('../util/class-attributes-helper');
+
 const config = require('config');
 const COOKIE_ACCESS_TOKEN = 'accessToken';
 
 const oauth2Route = (req, res, next) => {
   accessTokenRequest(req)
     .then(result => {
-      res.cookie(COOKIE_ACCESS_TOKEN, result.access_token,
-        {
-          maxAge: result.expires_in * 1000,
-          httpOnly: true,
-          secure: config.get('security.secure_auth_cookie_enabled')
+      if( hasAnAttribute(result,'access_token')) {
```

Comment:
more robust to rely on response codes rather than attributes in response payload.

Reply:
Yes  I agree with that . It should be rely on response.status as much as possible.   if( result.status === 200 ) . However the original code is been written to handle one  Promise from the response.json()  from accessTokenRequest().  Making the whole change to make it evaluate the response status  it will force the original implementation to handle 2 level of  Promises  .  This could introduce more risky code and a big refactor in test coverage for oauth2-route.js.